ewkcoffea/modules/corrections.py

```diff
@@ -144,65 +148,120 @@ def run3_muons_sf_Attach(muons,year,syst,id_method,iso_method):
     abseta_flat = ak.flatten(abs(muons.eta))
     pt_flat = ak.flatten(muons.pt)
 
-    # For now, cap all pt at 199.9 (limit for this particular sf)
+    # For now, cap all pt at 199.9 and min of 15.0 (limits for this particular json)
     pt_flat = ak.where(pt_flat>199.9,199.9,pt_flat)
     pt_flat = ak.where(pt_flat<15.0,15.0,pt_flat)
 
     # Evaluate the SF
     ceval = correctionlib.CorrectionSet.from_file(fname)
-    sf_id_flat = ceval[id_method].evaluate(abseta_flat,pt_flat,syst)
-    sf_iso_flat = ceval[iso_method].evaluate(abseta_flat,pt_flat,syst)
-    sf_flat = sf_id_flat * sf_iso_flat
-    sf = ak.unflatten(sf_flat,ak.num(muons.pt))
+    sf_id_flat_nom = ceval[id_method].evaluate(abseta_flat,pt_flat,"nominal")
+    sf_id_flat_syst = ceval[id_method].evaluate(abseta_flat,pt_flat,"syst")
+    sf_id_flat_hi = sf_id_flat_nom + sf_id_flat_syst
+    sf_id_flat_lo = sf_id_flat_nom - sf_id_flat_syst
+
+    sf_iso_flat_nom = ceval[iso_method].evaluate(abseta_flat,pt_flat,"nominal")
+    sf_iso_flat_syst = ceval[iso_method].evaluate(abseta_flat,pt_flat,"syst")
+    sf_iso_flat_hi = sf_iso_flat_nom + sf_iso_flat_syst
+    sf_iso_flat_lo = sf_iso_flat_nom - sf_iso_flat_syst
+
+    sf_flat_nom = sf_id_flat_nom * sf_iso_flat_nom
+    sf_flat_hi = sf_id_flat_hi * sf_iso_flat_hi
+    sf_flat_lo = sf_id_flat_lo * sf_iso_flat_lo
+
+    sf_nom = ak.unflatten(sf_flat_nom,ak.num(muons.pt))
```

Comment:
Flattening and unflattening can often be tricky. Just to make sure, could you confirm if at some point you've done some sanity checks (e.g. printing things out before and after flattening and unflattening) just to make sure things look like they're doing what you expect them to do, so you're confident that the flatten/unflatten is working as expected?

Reply:
Checked. Things seem to be flattening/unflattening properly